### ISSUE TITLE
Test loss stops improving.

### ISSUE THREAD

**Issue by jjjjeeerry0406** [KindXiaoming/pykan#535]
Hi all. I'm trying to use KAN to create a prediction model. Although the prediction accuracy is near 80%, I'm afraid it's an overfitting result. Here are the parameters of my model.
```
model = KAN(width=[8,10,7,4,3,3], grid=3, k=3,grid_eps=0,device=device)
model.fit(dataset,  opt="LBFGS",steps=step, update_grid=False,reg_metric='edge_forward_spline_n')
model.fit(dataset, opt="LBFGS", steps=step, update_grid=False,reg_metric='edge_forward_spline_n')
```
My train_loss can continue to reduce with more training iterations. , but my test_loss doesn't seem to be decreasing any further. Does this mean that my model training is complete?
```
saving model version 0.0
| train_loss: 8.71e-02 | test_loss: 1.14e-01 | reg: 8.15e+01 | : 100%|█| 30/30 [00:10<00:00,  2.76it
saving model version 0.1
| train_loss: 6.81e-02 | test_loss: 9.68e-02 | reg: 8.66e+01 | : 100%|█| 30/30 [00:10<00:00,  2.84it
saving model version 0.2
```
